packages/server-side-render/src/server-side-render.js

```diff
@@ -98,7 +98,7 @@ export default function ServerSideRender( props ) {
 		LoadingResponsePlaceholder = DefaultLoadingResponsePlaceholder,
 	} = props;
 
-	const isMountedRef = useRef( true );
+	const isMountedRef = useRef( false );
```

Comment:
But it did mount already, didn't it? Unless I misunderstand something, this looks wrong - can you elaborate?

Reply:
Setting the initial value doesn't mean that the component was mounted. React can call a component (function) for other reasons without mounting it.

This is a common (anti)pattern when you want to run synchronization only after the initial mount.

```js
const didMount = useRef( false );
useEffect( () => {
   didMount.current = true;
   return () => { didMount.current = false };
}, [] );

Reply:
@tyxla, I think this is the third issue I fixed for a similar pattern. I explained why the original implementations were failing in development mode here: https://github.com/WordPress/gutenberg/pull/62141#issuecomment-2139551221.

Original code:

```js
const didMount = useRef( true );
useEffect( () => {
   return () => { didMount.current = false };
}, [] );
```

Reply:
Ah, of course, got it. Wow, this component is messy 😉

Reply:
Yeah, it's bit of noodle soup 🍜